### ISSUE TITLE
Error in charge power when charging at IT 3phase

### ISSUE THREAD

**Issue by Allram** [balmli/com.tesla.charger#3]
Hello,

On the Charge power value it says my car is charging with ~8.5kW, i charge on a IT 3Phase grid so the correct should be more in the 3kW range. I know Tesla has made an update so the cars shows the correct kW now, and so does Teslamate(selfhosted logger). Is it possible to check this and how is this calculated?
![Screenshot_20201117_204101_app homey](https://user-images.githubusercontent.com/5573149/99440019-447d1700-2916-11eb-8673-ee802035e78b.jpg)
![Screenshot_20201117_204608_com android chrome](https://user-images.githubusercontent.com/5573149/99440022-46df7100-2916-11eb-9b81-010d692c757a.jpg)

**Comment by balmli** [comment]
Thanks you reporting!

This is somwthing I'm aware of.  (I just have one phase myself)

Will try to fix it.

**Comment by balmli** [comment]
Power is now calculated as: current * voltage * phases:

E.g:   11.6 A * 240 V * 3 = 8352 W

Do you know how the calculation should be for 3 phase chargers ?

**Comment by Genghes** [comment]
> Power is now calculated as: current * voltage * phases:
> 
> 
> 
> E.g:   11.6 A * 240 V * 3 = 8352 W
> 
> 
> 
> Do you know how the calculation should be for 3 phase chargers ?

Heisann.

For 3fas skal man ikke bruke antall faser (3) men kvadratroten av antall faser. Altså 1.732

V * 1.732 * A

Eller 

V * √3 * A

**Comment by balmli** [comment]
240 * √3 * 11,6 = 4822 W..

Så forslag til hvordan man kommer frem til 3 kw da ?


Her er felt fra "charger state":

https://tesla-api.timdorr.com/vehicle/state/chargestate

```
{
  "response": {
    "battery_heater_on": false,
    "battery_level": 90,
    "battery_range": 224.47,
    "charge_current_request": 40,
    "charge_current_request_max": 40,
    "charge_enable_request": true,
    "charge_energy_added": 29.41,
    "charge_limit_soc": 90,
    "charge_limit_soc_max": 100,
    "charge_limit_soc_min": 50,
    "charge_limit_soc_std": 90,
    "charge_miles_added_ideal": 118.5,
    "charge_miles_added_rated": 95.0,
    "charge_port_cold_weather_mode": null,
    "charge_port_door_open": true,
    "charge_port_latch": "Engaged",
    "charge_rate": 0.0,
    "charge_to_max_range": false,
    "charger_actual_current": 0,
    "charger_phases": null,
    "charger_pilot_current": 40,
    "charger_power": 0,
    "charger_voltage": 0,
    "charging_state": "Complete",
    "conn_charge_cable": "SAE",
    "est_battery_range": 171.24,
    "fast_charger_brand": "<invalid>",
    "fast_charger_present": false,
    "fast_charger_type": "<invalid>",
    "ideal_battery_range": 280.59,
    "managed_charging_active": false,
    "managed_charging_start_time": null,
    "managed_charging_user_canceled": false,
    "max_range_charge_counter": 0,
    "minutes_to_full_charge": 0,
    "not_enough_power_to_heat": false,
    "scheduled_charging_pending": false,
    "scheduled_charging_start_time": null,
    "time_to_full_charge": 0.0,
    "timestamp": 1604977209418,
    "trip_charging": false,
    "usable_battery_level": 90,
    "user_charge_enable_request": null
  }
}
```

**Comment by Allram** [comment]
Charger_power skal gi korrekt effekt nå med de siste måneders oppdatering på bilene.

Min oppgir 3 i verdi når jeg lader med 3kW = 220V *12A *1.15
1.15 er grunnet lading med 3fas IT (kvasi trefase som noen kaller det) ikke gir full 3Fas med mindre man har Model 3/Y og 230V ITnett. Men gir ca 15% høyere effekt enn 1faslading.

**Comment by balmli** [comment]
OK, da blir det jo greit.  Takk!

**Comment by Genghes** [comment]
Charger_phases skal vel kunne brukes til å indikere en eller trefas?

Men dersom charger_power er effekt i KW blir det jo enkelt ja.

Charger_phases brukes sikkert for å vise det lille 3 tallet i displayet i bilen når trefas er i bruk.

**Comment by balmli** [comment]
Check v. 1.0.32.

And please test :-)

**Comment by balmli** [comment]
A couple of fixes, so now you can look for 1.0.34.

**Comment by Allram** [comment]
This seems to work now on IT 3phase and also 1Phase. Haven't testet on 400V 3Phase, but i can't see why it shouldnt work there :)
